File: Source/AlphaTab/ProgressEventArgs.cs
namespace AlphaTab
{
    /// <summary>
    /// Represents the progress of any data being loaded.
    /// </summary>
    public class ProgressEventArgs
    {
        /// <summary>
        /// Gets the currently loaded bytes.
        /// </summary>
        public int Loaded { get; }

        /// <summary>
        /// Gets the total number of bytes to load.
        /// </summary>
        public int Total { get; }

        /// <summary>
        /// Initializes a new instance of the <see cref="ProgressEventArgs"/> class.
        /// </summary>
        /// <param name="loaded"></param>
        /// <param name="total"></param>
        public ProgressEventArgs(int loaded, int total)
        {
            Loaded = loaded;
            Total = total;
        }
    }
}

File: src/ProgressEventArgs.ts
/**
 * Represents the progress of any data being loaded.
 */
export class ProgressEventArgs {
    /**
     * Gets the currently loaded bytes.
     */
    public readonly loaded: number;

    /**
     * Gets the total number of bytes to load.
     */
    public readonly total: number;

    /**
     * Initializes a new instance of the {@link ProgressEventArgs} class.
     * @param loaded
     * @param total
     */
    public constructor(loaded: number, total: number) {
        this.loaded = loaded;
        this.total = total;
    }
}
